js/ui/viewSelector.js

```diff
@@ -51,14 +51,47 @@ const ViewsDisplayPage = {
     SEARCH: 2,
 };
 
+/** DiscoveryFeedButton:
+ *
+ * This class handles the button to launch the discovery feed application
+ */
+const DiscoveryFeedButton = new Lang.Class({
+    Name: 'DiscoveryFeedButton',
+    Extends: St.Button,
+
+    _init: function() {
+        let iconFile = Gio.File.new_for_uri('resource:///org/gnome/shell/theme/discovery-feed-open-tab.png');
+        let gicon = new Gio.FileIcon({ file: iconFile });
+        this._icon = new St.Icon({ gicon: gicon,
+                                   style_class: 'discovery-feed-icon',
+                                   track_hover: true });
+        this.parent({ name: 'discovery-feed',
+                      child: this._icon,
+                      x_align: Clutter.ActorAlign.CENTER,
+                      y_align: Clutter.ActorAlign.CENTER });
+    }
+});
+
+function allocateDiscoveryFeedButtonToBox(discoveryFeedButton, box, availWidth, flags) {
+    let discoveryFeedButtonHeight = discoveryFeedButton.get_preferred_height(availWidth)[1];
+    let discoveryFeedButtonBox = box.copy();
+    let x1 = (availWidth - discoveryFeedButton.get_width()) * 0.5;
+    discoveryFeedButtonBox.y1 = 0;
```

Comment:
Previously there was a class member being used for this, but it was always zero from what I could tell and never changed, so I just folded it in as a constant.